src/Compilers/CSharp/Test/Syntax/Parsing/FuzzTesting.cs

```diff
@@ -79,20 +79,42 @@ public void c024928()
             var tree = SyntaxFactory.ParseSyntaxTree(test);
         }
 
-        [WorkItem(2771, "https://github.com/dotnet/roslyn/issues/2771")]
-        [Fact]
-        public void TestBinary()
+        public void TestBinaryWithSeed(int seed)
         {
-            // use a fixed seed so the test is reproducible
-            var random = new System.Random(12345);
-            const int n = 40 * 1000 * 1000; // 40 million "character"s
-            var builder = new StringBuilder(n + 10);
-            for (int i = 0; i < n; i++)
+            var random = new System.Random(seed);
+            // 40 million "character"s
+            const int n = 40 * 1000 * 1000;
+            var str = new string(' ', n);
+
+            unsafe
+            {
+                fixed(char* chars = str)
+                {
+                    for(int i = 0; i < n; i++)
+                    {
+                        chars[i] = (char)random.Next(char.MaxValue);
+                    }
+                }
+            }
+
+            try {
+                var tree = CSharpSyntaxTree.ParseText(str);
+            }
+            catch (System.Exception e)
             {
-                builder.Append((char)random.Next(0xffff));
+                throw new System.Exception($"Failed with seed {seed}", e);
             }
-            var source = builder.ToString();
-            var tree = CSharpSyntaxTree.ParseText(source);
+        }
+
+        [WorkItem(2771, "https://github.com/dotnet/roslyn/issues/2771")]
+        [Fact]
+        public void TestBinaryRandom()
+        {
+            // If this test fails, make a new test method that
+            // calls TestBinaryWithSeed with the seed that failed.
+            var seed = new System.Random().Next();
```

Comment:
Might want to consider using [RNGCryptoServiceProvider ](https://msdn.microsoft.com/en-us/library/system.security.cryptography.rngcryptoserviceprovider(v=vs.110).aspx) as it will provide much better randomness over System.Random